src/actions/channels/CHANNEL_PINS_UPDATE.ts

```diff
@@ -0,0 +1,40 @@
+import { Action } from '../../lib/structures/Action';
+import { isTextBasedChannel } from '../../util/Util';
+
+import type { ChannelPinsUpdateDispatch } from '@klasa/ws';
+
+export default class CoreAction extends Action {
+
+	/**
+	 * Processes the event data from the websocket.
+	 * @since 0.0.1
+	 * @param data The raw data from {@link Client#ws}
+	 */
+	public run(data: ChannelPinsUpdateDispatch): void {
+		const guild = (data.d.guild_id && this.client.guilds.get(data.d.guild_id)) ?? null;
+		const channel = guild ? guild.channels.get(data.d.channel_id) : this.client.channels.get(data.d.channel_id);
+		if (!channel || !isTextBasedChannel(channel)) return;
```

Comment:
Isn't the second check... unnecessary? You can't have pins on a voice channel

Reply:
Type and safe guard against any unexpected data Discord may give.